message-markdown.md

```diff
@@ -0,0 +1,44 @@
+# Markdown in messages
+
+Deltachat is going to support the following subset of markdown:
+
+### `**bold**`
+
+### `__underline__`
+
+### `~~strikethrough~~`
+
+### `https://delta.chat` - Urls
+
+Make URLs clickable, but **don't** support the Link syntax `[Name](url)`
+because that could be used for misleading and as such be used in scams.
+
+### `` `inline-code` ``
+useful to send non markdown text in your message like source code snippets.
+Should get rendered in a monospace font and with a different background.
+ 
+### ` ``` fence code block ``` `
+
+```md
+Similar to `inline-code` but not inline and it may supports code highlighting.
+```
+` ```[lang?] [content]``` `
+ A bit modified from the common syntax to allow one liners.
```

Comment:
` ```[lang?] [content]``` ` is to mark a line of c/rust/java/whatever code? this seems to be a quite tricky part and/or needs a lib that supports that. not sure if this is really needed in every day use for most users but it raises the barrier of supporting markdown if we say "Deltachat is going to support the following subset".

i'd suggest to strike that.

Reply:
I'd suggest marking the actual syntax highlighting as optional (if the client doesn't want to support it it can just ignore the lang definition.

Reply:
and the content has to be put in square brackets? or as in https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting ? however, as mentioned, i would not add this to the spec in this early level anyway :)

Reply:
no the square brackets are my description of a variable here

Reply:
I was stumbling over the square brackets too, because I thought thery're part of the syntax. Maybe you can remove them in the example to avoid confusion?

Reply:
what about the questionmark then? is that also part of the syntax?

Reply:
githubs editor doesn't work currently so maybe we can write it like
` ```language content``` ` or ` ``` content``` `

Reply:
also there is a question how to go about no language definition, At the moment I think one space before the content is fine, but not optimal. We could also go no language support in single line mode